File: chapter01/aggregation/example-1.4.1.3/ex1413.ml
let floats_sum floats =
  let rec sum_acc acc = function
    | [] -> acc
    | hd::tl -> sum_acc (acc +. hd) tl
  in
  sum_acc 0. floats

List.range 1 11
|> List.filter ~f:(fun k -> Float.of_int k |> Float.cos |> Float.compare 0. < 0)
|> List.map ~f:(fun k -> Float.of_int k |> Float.cos |> Float.sqrt)
|> floats_sum

File: chapter01/aggregation/example-1.4.1.3/ex1413.rs
fn main() {
    let ans = (1..=10)
        .filter(|k| f64::cos(*k as f64) > 0f64)
        .map(|k| f64::sqrt(f64::cos(k as f64)))
        .sum::<f64>();

    println!("{}", ans);
}
